manifests/install.pp

```diff
@@ -121,6 +133,41 @@
         }
         Yumrepo['influxdata'] -> Package[$telegraf::package_name]
       }
+      elsif $telegraf::manage_archive {
```

Comment:
I did think about that when adding this, but didn't really want to rewrite a bunch of the logic.  I added the warnings for unsupported install methods, because I tried to install from archive on CentOS and it just fell through the `if` statement without installing any packages.